### DISCUSSION TITLE
How to render zsh prompt color escape sequences?

### DISCUSSION THREAD

**Discussion by hongyilyu** [starship/starship#6406] [Q&A] [answered]
Hi everyone,

I'm trying to customize my Zsh prompt and use color escape sequences to display text with colors (The text is generated from gitstatusd):

```
 %76Fmaster %178F!3%f
```

the output was colored back in starship `1.16.0`, but it turns to this pure text after upgrading to `1.21.1`

Any insights or suggestions on how to properly render this sequence with colors would be greatly appreciated!

Thanks in advance!

**Comment by davidkna** [comment, marked as answer]
Since starship v1.20 shell-interpretable characters in custom command output are escaped unless `unsafe_no_escape` is enabled. Please try enabling the option for the module, or configure `gitstatusd` not to use the escapes.

Please keep in mind that if `gitstatusd` does not take care to escape its output, enabling the option could also lead to code execution.
